translator/opencensus/oc_to_internal.go

```diff
@@ -0,0 +1,388 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensus
+
+import (
+	"strings"
+
+	occommon "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	octrace "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+func ocToInternal(td consumerdata.TraceData) data.TraceData {
+
+	if td.Node == nil && td.Resource == nil && len(td.Spans) == 0 {
+		return data.TraceData{}
+	}
+
+	resource := ocNodeResourceToInternal(td.Node, td.Resource)
+
+	resourceSpans := data.NewResourceSpans(resource, nil)
+	resourceSpanList := []*data.ResourceSpans{resourceSpans}
+
+	spanCount := len(td.Spans)
+
+	if spanCount != 0 {
```

Comment:
To reduce indentation / nesting levels, this condition could be reversed:
```
if spanCount == 0 {
  return data.NewTraceData(resourceSpanList)
}

...
```
or, alternatively, the loop below can be extracted into a separate function.

Reply:
Done